casper/src/test/scala/coop/rchain/casper/util/rholang/DeployerAuthTest.scala

```diff
@@ -0,0 +1,117 @@
+package coop.rchain.casper.util.rholang
+
+import cats.effect.Resource
+import com.google.protobuf.ByteString
+import coop.rchain.casper.util.ConstructDeploy
+import coop.rchain.casper.util.rholang.Resources._
+import coop.rchain.crypto.PrivateKey
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.signatures.Ed25519
+import coop.rchain.models.Expr.ExprInstance.GBool
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.models.{GDeployerAuth, Par}
+import coop.rchain.rholang.interpreter.accounting
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.concurrent.duration._
+
+class DeployerAuthTest extends FlatSpec with Matchers {
+
+  private val runtimeManager: Resource[Task, RuntimeManager[Task]] =
+    mkRuntimeManager("deployer-auth-runtime-manager-test")
+
+  "Deployer authentication" should "be forwarded to the reducer" in {
+    val sk = PrivateKey(
+      Base16.unsafeDecode("b18e1d0045995ec3d010c387ccfeb984d783af8fbb0f40fa7db126d889f6dadd")
+    )
+    val pk             = ByteString.copyFrom(Ed25519.toPublic(sk).bytes)
+    val captureChannel = "__DEPLOYER_AUTH_VALUE__"
+    val deploy = ConstructDeploy.sourceDeploy(
+      source = s"""new auth(`rho:deployer:auth`) in { @"$captureChannel"!(*auth) }""",
+      timestamp = System.currentTimeMillis(),
+      accounting.MAX_VALUE,
+      sec = sk
+    )
+    val result =
+      runtimeManager
+        .use(mgr => mgr.captureResults(mgr.emptyStateHash, deploy, captureChannel))
+        .runSyncUnsafe(10.seconds)
+
+    result.size should be(1)
+    result.head should be(GDeployerAuth(pk): Par)
+  }
+
+  it should "make drain vault attacks impossible" in {
+    val skDeployer = PrivateKey(
+      Base16.unsafeDecode("b18e1d0045995ec3d010c387ccfeb984d783af8fbb0f40fa7db126d889f6dadd")
+    )
+    val skAttacker = PrivateKey(
+      Base16.unsafeDecode("c18e1d0045995ec3d010c387ccfeb984d783af8fbb0f40fa7db126d889f6dadd")
+    )
+    val deploy = ConstructDeploy.sourceDeploy(
+      source =
+        s"""new auth(`rho:deployer:auth`) in { contract @"checkAuth"(input, ret) = { ret!(*input == *auth) }}""",
+      timestamp = System.currentTimeMillis(),
+      accounting.MAX_VALUE,
+      sec = skDeployer
+    )
```

Comment:
Please extract a `deploy(deployer, rho)` private method in this test that hides the irrelevant timestamp and phloLimmimt parameters

Reply:
Done